File: SupplierPortal.Web/Modules/ServerTypes/Market/OfferSupplierForm.ts
import { LookupEditor, IntegerEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface OfferSupplierForm {
    SupplierId: LookupEditor;
    OfferId: IntegerEditor;
    Email: StringEditor;
    OfficalName: StringEditor;
    OfficalSurname: StringEditor;
    CallNumber: StringEditor;
}

export class OfferSupplierForm extends PrefixedContext {
    static readonly formKey = 'Market.OfferSupplier';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!OfferSupplierForm.init)  {
            OfferSupplierForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = StringEditor;

            initFormType(OfferSupplierForm, [
                'SupplierId', w0,
                'OfferId', w1,
                'Email', w2,
                'OfficalName', w2,
                'OfficalSurname', w2,
                'CallNumber', w2
            ]);
        }
    }
}

File: SupplierPortal.Web/Modules/Market/OfferSupplier/OfferSupplierForm.cs
using MovieTutorial.MovieDB;
using Serenity.ComponentModel;

namespace SupplierPortal.Market.Forms;

[FormScript("Market.OfferSupplier")]
[BasedOnRow(typeof(OfferSupplierRow), CheckNames = true)]
public class OfferSupplierForm
{
    public int SupplierId { get; set; }
    [Visible(false)]
    public int OfferId { get; set; }
    public string Email { get; set; }
    public string OfficalName { get; set; }
    public string OfficalSurname { get; set; }
    public string CallNumber { get; set; }

}